services/agents/jobs.go

```diff
@@ -18,30 +18,283 @@
 package agents
 
 import (
+	"context"
 	"time"
 
 	"github.com/percona/pmm/api/agentpb"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/types/known/durationpb"
 	"gopkg.in/reform.v1"
 
 	"github.com/percona/pmm-managed/models"
 )
 
+// ErrRetriesExhausted is returned when remaining retries are 0.
+var ErrRetriesExhausted = errors.New("Retries exhausted")
+
 // JobsService provides methods for managing jobs.
 type JobsService struct {
 	r  *Registry
 	db *reform.DB
+
+	retentionService retentionService
+	l                *logrus.Entry
 }
 
 // NewJobsService returns new jobs service.
-func NewJobsService(db *reform.DB, registry *Registry) *JobsService {
+func NewJobsService(db *reform.DB, registry *Registry, retention retentionService) *JobsService {
 	return &JobsService{
-		r:  registry,
-		db: db,
+		db:               db,
+		r:                registry,
+		retentionService: retention,
+		l:                logrus.WithField("component", "agents/jobsService"),
 	}
 }
 
+func (s *JobsService) RestartJob(ctx context.Context, jobID string) error {
```

Comment:
So, we reuse the same job entry right? won't we loose any data after retry?

Reply:
We can lose last error. It will be overriden.